src/extract.js

```diff
@@ -0,0 +1,32 @@
+import glob from 'glob'
+import ProgressBar from 'progress'
+import { transformFileSync } from 'babel-core'
+import babelGettextExtractor from './babel-gettext-extractor'
+
+const defaultBabelConfig = {
+  presets: ['react', 'stage-2'],
+}
+
+export default (fileGlobs, getConfig, showProgress = false) => {
+  const files = glob.sync(process.argv[2])
+
+  const progressBar = showProgress
+    ? new ProgressBar(' extracting [:bar] :percent :fileName', {
+      total: files.length,
+      width: 10,
+    })
+    : { tick: () => {} }
+
+  files.forEach((fileName) => {
```

Comment:
I would only extract this `forEach` block into an API function. I think for an API you rather want to allow passing just an array of files instead of a glob pattern. Also showing the ProgressBar does not make sense for the API scenario.